Library/Homebrew/cmd/migrate-linux-core.rb

```diff
@@ -0,0 +1,48 @@
+# typed: false
+# frozen_string_literal: true
+
+require "utils/popen"
+
+module Homebrew
+  extend T::Sig
+
+  MIGRATION_CACHE_FILE = (HOMEBREW_CACHE/"linuxbrew-core-migration-done").freeze
+
+  module_function
+
+  def migrate_linux_core
+    # Don't run the migration again if it is already done
+    return if File.exist?(MIGRATION_CACHE_FILE)
+
+    formulae_to_migrate = []
+
+    # TODO: Fill final list below just before the brew release
+    linuxbrew_core_versions =
+      {
+        "afuse"    => { "version_revision" => "0.4.1_2" },
+        "apng2gif" => { "version_revision" => "1.8_2" },
+        "apr-util" => { "version_revision" => "1.6.1_5" },
+      }
+
+    installed = Utils.safe_popen_read(
+      ENV["HOMEBREW_BREW_FILE"], "list", "--formula", "--versions"
+    ).split("\n")
+    installed.each do |formula_version|
+      formula, version_revision = formula_version.split
+
+      if linuxbrew_core_versions.include?(formula) &&
+         version_revision == linuxbrew_core_versions[formula]["version_revision"]
+        formulae_to_migrate << formula
+      end
+    end
+
+    return if formulae_to_migrate.empty?
+
+    ohai "Migrating linuxbrew-core to homebrew-core, this might take some time..."
+    system ENV["HOMEBREW_BREW_FILE"], "uninstall", "--force", "--ignore-dependencies", *formulae_to_migrate
+    system ENV["HOMEBREW_BREW_FILE"], "cleanup", *formulae_to_migrate
```

Comment:
We need this to not be done if `HOMEBREW_NO_INSTALL_CLEANUP` is set.

Reply:
We need to delete the linux bottle from the cache, because if the revision is bumped later on in homebrew-core; the wrong file will be on disk and there will be a sha256 mismatch.

Reply:
We cannot rely on `cleanup` to do that in that case, it does other things that the user may not wish to be done.